accord-core/src/main/java/accord/coordinate/CheckOn.java

```diff
@@ -136,21 +138,28 @@ class OnDone implements MapReduceConsume<SafeCommandStore, Void>
 
         public OnDone()
         {
-            Ranges localRanges = node.topology().localRangesForEpochs(txnId.epoch, untilLocalEpoch);
-            PartialRoute<?> selfRoute = route().slice(localRanges);
+            Ranges sliceRanges = node.topology().localRangesForEpochs(txnId.epoch(), untilLocalEpoch);
+            PartialRoute<?> selfRoute = route().slice(sliceRanges, Maximal);
```

Comment:
So this one is a little interesting, and I am open to suggestions. Essentially this is to make our invariants happy for ranges, but it could lead to fetching more data than we want. Basically I choose never to slice a `Range` that is one of the keys of a transaction to ensure identity is never broken. But when we come to reconstitute a transaction, we don't fully cover the keys. Perhaps we should modify the invariants of `reconstitutePartial` and the objects that make it up. I will take another look to refresh my memory.

Reply:
Ok, so this `Maximal` wasn't necessary, and in fact getting a `Maximal` _`Route`_ is probably nonsensical and dangerous so I have modified `slice` to return only a basic collection (non-Route) if a specific `Slice` is requested. I haven't removed `Maximal` as a feature for now as it might be easier to use this later inside of `PartialTxn` etc. when either slicing or reconstituting to define `covers`, though open to deleting it.